client/components/content-elements/tce-pdf/edit/Toolbar.vue

```diff
@@ -31,9 +31,13 @@ export default {
     url: vm => get(vm.element, 'data.assets.url')
   },
   methods: {
-    save({ url, publicUrl }) {
+    save({ url }) {
       const element = cloneDeep(this.element);
-      set(element.data, 'assets.url', url);
+      if (url) set(element.data, 'assets.url', url);
+      else {
+        set(element.data, 'assets', null);
+        set(element.data, 'url', null);
```

Comment:
Since data.url is virtual prop it might make sense to update patch action on the content element ctrl to refetch. That will remove the need for cleaning up virtual props (data.url in this case).